### PR TITLE
Build for only one target for docs.rs

### LINKED ISSUE #5378: docs.rs request: only build for one target
This project has a lot of crates and as a result it takes docs.rs a while to build all of them (we're [currently at 107 queued](https://docs.rs/releases/queue)). Do you mind opting-in to only building one target? That would allow us to decrease build times for both your project and other crates in the queue.

For your project, that would look like
```toml
[package.metadata.docs.rs]
targets = ["x86_64-unknown-linux-gnu"]
```
in your Cargo.toml, which says to build exactly one target, x86_64 linux, and use that as the homepage for your docs.  Your project is currently failing on targets besides x86_64-unknown-linux-gnu in any case (because of `ring`), so it won't affect the documentation that's displayed at all, only your build times.

See https://blog.rust-lang.org/2020/03/15/docs-rs-opt-into-fewer-targets.html for more details about opting into fewer targets.

### PR DESCRIPTION
**Pull request by pscott**
This PR specifies that we wish to only build for one target for docs.rs.
This will drastically reduce build time for docs.rs.

Closes #5378

### CODE REVIEW

**Comment by bkchr** [comment]
You will probably need to set this option for each crate.

**Comment by pscott** [comment]
> You will probably need to set this option for each crate.

I just ran this script:
```sh
CRATES=$(find frame client primitives utils -name "Cargo.toml")
for cargo in $CRATES
do
	echo "\n[package.metadata.docs.rs]\ntargets = ["x86_64-unknown-linux-gnu"]" >> $cargo
done
```
Which adds the metadata line in each `Cargo.toml` file in the folders `frame`, `client`, `primitives`, and `utils`.

This means there are some unpublished crates to which I've added the metadata. Since I have not found a better way of doing this (unless using crates.io's API), I think it's ok.

If this is not ok, and we really wish to add the docs.rs metadata to ONLY crates that have been published, I can write a better script. Thoughts @bkchr ?

**Comment by pscott** [comment]
For some reason lots of different reviewers got asked for this PR. I think there's a magic shortcut that I pressed that asked a review from all code owners. Sorry for this unnecessary ping.

File: client/Cargo.toml [hunk 1]
```diff
@@ -43,3 +43,9 @@ tempfile = "3.1.0"
 substrate-test-runtime-client = { version = "2.0.0-dev", path = "../test-utils/runtime/client" }
 kvdb-memorydb = "0.5.0"
 sp-panic-handler = { version = "2.0.0-alpha.5", path = "../primitives/panic-handler" }
+
+[package.metadata.docs.rs]
+targets = [x86_64-unknown-linux-gnu]
+
+[package.metadata.docs.rs]
+targets = [x86_64-unknown-linux-gnu]
```
Comment (gnunicorn) [inline client/Cargo.toml]: seems like you ran it twice ... maybe you want to checkout from zero and redo?!?

**Comment by bkchr** [comment]
Did you even try to compile it locally? Everything fails.

**Comment by pscott** [comment]
> Did you even try to compile it locally? Everything fails.

Fixed. I had not compiled it locally, and indeed I was missing some escape characters in my script :)

**Comment by bkchr** [comment]
@gnunicorn is that something that should be integrated in your tool? The pr looks good and can be merged, but this should be automated in the future.

**Comment by gnunicorn** [comment]
> is that something that should be integrated in your tool? The pr looks good and can be merged, but this should be automated in the future.

You mean `cargo-unleash`? Yeah, probably.Also wanted to add a `patch`-section manager ...

**Comment by gnunicorn** [comment]
CI failures are unrelated.

**Comment by bkchr** [comment]
Yeah I meant cargo unleash

**Pull request merged**